modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java

```diff
@@ -44,78 +51,205 @@ public class OrchestratorApi {
-     * Unregister registered managed service.
+     * Update registered service using id and ocl file url.
      *
-     * @param managedServiceName name of managed service
+     * @param id          id of registered service.
+     * @param oclLocation URL of new Ocl.
      * @return response
      */
-    @Tag(name = "Service Vendor",
-            description = "APIs to publish the managed services.")
-    @DeleteMapping("/register/{managedServiceName}")
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs to manage register services.")
+    @Operation(description = "Update registered service using id and ocl file url.")
+    @PutMapping(value = "/register/file/{id}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
     @Transactional
-    public Response unregister(@PathVariable("managedServiceName") String managedServiceName) {
-        log.info("Unregistering managed service with name {}", managedServiceName);
-        this.orchestratorService.unregisterManagedService(managedServiceName);
+    public Response fetchUpdate(@PathVariable(name = "id") String id,
+            @RequestParam(name = "oclLocation") String oclLocation) throws Exception {
+        log.info("Update registered service {} with Url {}", id, oclLocation);
+        registerService.updateRegisteredServiceByUrl(id, oclLocation);
         String successMsg = String.format(
-                "Managed service %s unregistered success.", managedServiceName);
+                "Update registered service %s with Url %s", id, oclLocation);
+        log.info(successMsg);
         return Response.successResponse(successMsg);
     }
 
+    /**
+     * Unregister registered service using id.
+     *
+     * @param id id of registered service.
+     * @return response
+     */
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs unregister registered service.")
+    @Operation(description = "Unregister registered service using id.")
+    @DeleteMapping("/register/service/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    @Transactional
+    public Response unregister(@PathVariable("id") String id) {
+        log.info("Unregister registered service using id {}", id);
+        registerService.unregisterService(id);
+        String successMsg = String.format(
+                "Unregister registered service using id %s success.", id);
+        log.info(successMsg);
+        return Response.successResponse(successMsg);
+    }
+
+
+    /**
+     * List registered service with query params.
+     *
+     * @param cspName     name of cloud service provider.
+     * @param serviceName name of registered service.
+     * @return response
+     */
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs to list the managed services.")
+    @Operation(description = "List registered service with query params.")
+    @GetMapping(value = "/register/services",
```

Comment:
```suggestion
    @GetMapping(value = "/register",
```
Did `/register` works here?